src/app/control-planes/views/ControlPlaneDetailView.vue

```diff
@@ -13,111 +13,166 @@
         </h1>
       </template>
 
-      <div
-        class="stack"
-        data-testid="detail-view-details"
+      <DataSource
+        v-slot="{ data: globalInsight, error: globalInsightError }: GlobalInsightSource"
+        src="/global-insight"
       >
-        <ControlPlaneStatus />
-
-        <div class="columns">
-          <KCard v-if="can('use zones')">
-            <template #body>
-              <DataSource
-                v-slot="{ data, error }: ZoneOverviewCollectionSource"
-                src="/zone-cps?page=1&size=10"
+        <DataSource
+          v-slot="{ data: meshInsightsData, error: meshInsightsError }: MeshInsightCollectionSource"
+          src="/mesh-insights?page=1&size=10"
+        >
+          <DataSource
+            v-slot="{ data: zoneOverviewsData, error: zoneOverviewsError }: ZoneOverviewCollectionSource"
+            src="/zone-cps?page=1&size=10"
+          >
+            <ErrorBlock
+              v-if="globalInsightError"
+              :error="globalInsightError"
+            />
+
+            <LoadingBlock v-else-if="globalInsight === undefined" />
+
+            <template v-else>
+              <div
+                class="stack"
+                data-testid="detail-view-details"
               >
-                <ErrorBlock
-                  v-if="error"
-                  :error="error"
-                />
+                <template v-if="globalInsight.meshes.total === 1 && meshInsightsData?.items[0].name === 'default' && globalInsight.dataplanes.standard.total === 0">
```

Comment:
What is the difference between the `globalInsight.meshes.total` and the `meshInsightsData.total`?

Reply:
There shouldn’t be any.

Reply:
I thought we agreed on just looking at number of meshes == 1 ? This would enable to avoid the triple nesting of datasources and much quicker 1st frame rendering no? It would also make the code a LOT more readable I believe

Reply:
I think checking for there being exactly one mesh is enough. The check for this is now simpler and only requires moving the GlobalInsight data source a bit further up in the relevant component.